modules/apps/fragment/fragment-impl/src/main/java/com/liferay/fragment/internal/renderer/ContentObjectFragmentRenderer.java

```diff
@@ -130,8 +130,8 @@ public void render(
 		Tuple tuple = _getTuple(
 			displayObject.getClass(), fragmentRendererContext);
 
-		InfoItemRenderer infoItemRenderer = (InfoItemRenderer)tuple.getObject(
-			0);
+		InfoItemRenderer<Object> infoItemRenderer =
+			(InfoItemRenderer<Object>)tuple.getObject(0);
```

Comment:
Hey @hhuijser,

Why is this casting necessary? Isn't it redundant?
I've noticed that you've added many of them like this.

Reply:
@JorgeFerrer, I didn't add the typecase here. It was already there, I only added the generics. We do need to typecast when calling `Tuple.getObject` since that one returns `Object`.

For other cases where I did add typecast, it was in order to keep the compiler happy.

The reason for forcing the generics is to prevent sloppy and at times just plain bad implementation. I found quite a few while working on this. See also https://github.com/liferay/liferay-portal/blob/master/modules/util/source-formatter/src/main/resources/documentation/checks/generic_type_check.markdown

Reply:
Got it. 
Thanks for the explanation.